### ISSUE TITLE
Food spoil

### ISSUE THREAD

**Issue by noone57** [A2Bcorp/RW_A2B#35]
Right now, food doesn't spoil on the belts. Which is not the best, as this can be exploited in a too easy way. 

It would be best if food on the belt would spoiled like any other.

**Comment by TehJoE** [comment]
Fixed in 8dd733e45fe088571ec347ca1bec8b5bc6d2126e

**Comment by noone57** [comment]
This is really great. One of the last major issues in the mod is now resolved. I'm curious to see what will happen for items on the Undercovers, when under walls ... I'll have to do a test play to check it out !

**Comment by TehJoE** [comment]
If I had to guess, they probably fall back to deteriorating/rotting at the
outside temperature rate, since if I recall correctly, hovering over a wall
will show the outdoor temperature for that cell.

The food spoiling was a pretty simple fix, just calling `TickRare()` on the
contents of the belt every 250 ticks to update a food item's `CompRotting`.
`ItemContainer`s pass normal ticks through but for some reason don't do
that for rare ticks.

As far as item deterioration is concerned, that was a little more
complicated. I basically had to implement my own version of the normal
deterioration code that applied to belts' items. It was also the first use
I can recall of the reflection library I wrote in `GeneralUtilities`. I
knew that'd come in handy for some pesky private function that needed
calling ;)
